docs(blog): add dbt-ibis post

binste opened the pull request:
Based on [this comment](https://github.com/ibis-project/ibis/issues/7531#issuecomment-1823300792) by @lostmygithubaccount, this PR adds a new blog post on [dbt-ibis](https://github.com/binste/dbt-ibis). Thanks for the opportunity, this is much appreciated! I think the combination of dbt and Ibis is very powerful and I'm glad if we can make more people aware of the possibilities :)

So far it's only a very rough outline. @lostmygithubaccount any inputs/guidelines at this stage or ok if I expand this into a full post over the next few days?

lostmygithubaccount commented:
I'll take a deeper look on Monday but feel free to expand into a full post! I like the outline and am excited to see this posted, I think dbt and Ibis go great together

binste commented:
Great! :) Expanded the post and it's ready to review.

lostmygithubaccount commented on docs/posts/dbt-ibis/index.qmd:
```diff
@@ -0,0 +1,93 @@
+
+
+# Why dbt and Ibis go great together
+[dbt-ibis](https://github.com/binste/dbt-ibis) offers a more lightweight and compatible alternative, which allows you to write dbt models using Ibis. dbt-ibis transparently converts your Ibis statements into SQL and then hands it over to dbt. Your database does not need to have Python support for this as everything is executed in the same process as dbt. Rewriting the above SQL model in Ibis we g ...
```
would expand here and emphasize the point -- this allows for working in Python for all dbt adapters with supported Ibis backends (nearly all?) -- this is a great point!

lostmygithubaccount commented on docs/posts/dbt-ibis/index.qmd:
```diff
@@ -0,0 +1,93 @@
+
+Using Ibis instead of SQL for dbt models brings you many advantages:
+
+* Type checks and validation before your code is even executed in a database
```
nit: would remove "even"

lostmygithubaccount commented on docs/posts/dbt-ibis/index.qmd:
```diff
@@ -0,0 +1,93 @@
+
+* Type checks and validation before your code is even executed in a database
+* More composable as you can break down complex queries into smaller pieces
+* Better reusability of code. Although dbt allows you to use [Jinja and macros](https://docs.getdbt.com/docs/build/jinja-macros), which is an improvement over plain SQL, this gets you only so far. String manipulation is inherently fragile. With dbt-ibis, you can easily share common code between models.
```
I never liked the use of Jinja :)

binste replied:
Same ;)

lostmygithubaccount commented on docs/posts/dbt-ibis/index.qmd:
```diff
@@ -0,0 +1,93 @@
+* Type checks and validation before your code is even executed in a database
+* More composable as you can break down complex queries into smaller pieces
+* Better reusability of code. Although dbt allows you to use [Jinja and macros](https://docs.getdbt.com/docs/build/jinja-macros), which is an improvement over plain SQL, this gets you only so far. String manipulation is inherently fragile. With dbt-ibis, you can easily share common code between models.
+* Your dbt models become backend agnostic which reduces lock-in to a specific database. Furthermore, you get the possibility of building a [multi-engine data stack](https://juhache.substack.com/p/n-engines-1-language?utm_source=post-email-title&publication_id=1211981&post_id=137718100). For example, you could use DuckDB for small to medium workloads and Snowflake for heavy workloads and as an end ...
```
remove the "?utm_source" part of this link

lostmygithubaccount commented on docs/posts/dbt-ibis/index.qmd:
```diff
@@ -0,0 +1,93 @@
+* More composable as you can break down complex queries into smaller pieces
+* Better reusability of code. Although dbt allows you to use [Jinja and macros](https://docs.getdbt.com/docs/build/jinja-macros), which is an improvement over plain SQL, this gets you only so far. String manipulation is inherently fragile. With dbt-ibis, you can easily share common code between models.
+* Your dbt models become backend agnostic which reduces lock-in to a specific database. Furthermore, you get the possibility of building a [multi-engine data stack](https://juhache.substack.com/p/n-engines-1-language?utm_source=post-email-title&publication_id=1211981&post_id=137718100). For example, you could use DuckDB for small to medium workloads and Snowflake for heavy workloads and as an end ...
+* Unit test your code with your favorite Python testing frameworks such as pytest.
```
this is pretty cool! they're working on adding unit testing to dbt now, but it will take a while -- this could be positioned as a good alternative in the meantime

binste replied:
Yeah I saw it on the new roadmap, got me excited but I was also sad that a proper plugin system is not on the horizon for the next few months... Fully agree that this is a great alternative in the meantime :)

lostmygithubaccount commented on docs/posts/dbt-ibis/index.qmd:
```diff
@@ -0,0 +1,93 @@
+
+In addition, you can stick to the tool (Ibis) you like, no matter if you're writing an ingestion pipeline, a dbt model to transform the data in your data warehouse, or conduct an ad-hoc analysis in a Jupyter notebook.
+
+Be aware that a current limitation of dbt-ibis is, that you cannot connect to the database from within your dbt models, i.e. you purely use Ibis to construct a SELECT statement. You cannot execute statements and act based on the results.
```
would removce the comma after "is"

lostmygithubaccount commented on docs/posts/dbt-ibis/index.qmd:
```diff
@@ -0,0 +1,93 @@
+
+Be aware that a current limitation of dbt-ibis is, that you cannot connect to the database from within your dbt models, i.e. you purely use Ibis to construct a SELECT statement. You cannot execute statements and act based on the results.
+
+# Further readings
```
might want a further call to action either here or below -- is there a good link to point them to for getting started w/ dbt-ibis and trying it out?

lostmygithubaccount reviewed (commented):
thanks for writing this! mostly some small nits

I believe the Ibis devs will want you to wrap lines in the markdown (your editor can probably do this), and will need to squash everything into one commit of the right format 

apologies for the delay in reviewing here

binste commented on docs/posts/dbt-ibis/index.qmd:
```diff
@@ -0,0 +1,93 @@
+---
+
+# Introduction to dbt
+[dbt](https://github.com/dbt-labs/dbt-core) has revolutionized how transformations are orchestrated and managed within modern data warehouses. Initially released in 2016 by Fishtown Analytics, now called dbt Labs, dbt quickly gained traction within the data community due to its focus on bringing software engineering best practices to analytics code like modularity, portability, CI/CD, and documen ...
```
Sounds good, done

binste commented:
Thanks for the great inputs @lostmygithubaccount ! I believe I've implemented them all but let me know if you spot anything else. I also wrapped the lines and squashed all commits into one which hopefully has a good commit message.
